src/app/launcher/create-app/mission-runtime-createapp-step/mission-runtime-createapp-step.component.ts

```diff
@@ -23,25 +55,31 @@ import { LauncherStep } from '../../launcher-step';
   styleUrls: ['./mission-runtime-createapp-step.component.less']
 })
 export class MissionRuntimeCreateappStepComponent extends LauncherStep implements OnInit, OnDestroy {
-  private _missions: Mission[] = [];
-  private _runtimes: Runtime[] = [];
+  private _missions: ViewMission[] = [];
+  private _runtimes: ViewRuntime[] = [];
+  private _boosters: Booster[] = null;
 
   private missionId: string;
   private runtimeId: string;
+  private versionId: string;
   private subscriptions: Subscription[] = [];
 
   constructor(@Host() public launcherComponent: LauncherComponent,
               private missionRuntimeService: MissionRuntimeService,
-              public _DomSanitizer: DomSanitizer) {
+              public _DomSanitizer: DomSanitizer,
+              private broadcaster: Broadcaster) {
     super();
   }
 
   ngOnInit() {
     this.launcherComponent.addStep(this);
-    this.subscriptions.push(this.missionRuntimeService.getMissions()
-      .subscribe(missions => this._missions = missions));
-      this.subscriptions.push(this.missionRuntimeService.getRuntimes()
-      .subscribe(runtimes => this._runtimes = runtimes));
+    this.subscriptions.push(this.missionRuntimeService.getBoosters()
+      .subscribe(boosters => {
+        this._boosters = boosters;
+        this.updateSelection();
+        this.restoreFromSummary();
+      }));
+    this.broadcaster.on('cluster').subscribe(() => this.updateSelection());
```

Comment:
You can do `this.broadcaster.on<Cluster>('cluster').subscribe(cluster => this.updateSelection(cluster));`?

Reply:
Same as previous